libshaderc_util/src/compiler_test.cc

```diff
@@ -187,4 +187,21 @@ TEST_F(CompilerTest, DISABLED_RespectTargetEnvOnVulkanShader) {
   // TODO(dneto): Add test for a shader that should only compile for Vulkan.
 }
 
+TEST_F(CompilerTest, AddMacroDefinition) {
+  const std::string kMinimalExpandedShader = "void E(){}";
+  compiler_.AddMacroDefinition("E", 1u, "main", 4u);
```

Comment:
Also add a test where the length matters for both arguments, i.e. the passed-in length is *smaller* than the length of the char* argument.

E.g. 
  AddMacroDefinition("EFGH", 1u, "mainly", 4u);